### PR TITLE
CSHARP-5306: Make MongoDatabase, MongoCollection and MongoClient base internal

### PR DESCRIPTION
**Pull request by BorisDog**

### CODE REVIEW

File: tests/MongoDB.Driver.Tests/MongoDatabaseImplTests.cs [hunk 1]
```diff
@@ -1462,7 +1462,7 @@ private Mock<IMongoClient> CreateMockClient()
                     settings = settings ?? new MongoDatabaseSettings();
                     settings.ApplyDefaultValues(mockClient.Object.Settings);
                     var cluster = new Mock<IClusterInternal>().Object;
-                    return new MongoDatabaseImpl(mockClient.Object, databaseNamespace, settings, cluster, _operationExecutor);
+                    return new MongoDatabase(mockClient.Object, databaseNamespace, settings, cluster, _operationExecutor);
```
Comment (rstam) [inline tests/MongoDB.Driver.Tests/MongoDatabaseImplTests.cs:1465]: We should rename this file also to `MongoDatabaseTests.cs`.

**Review by rstam — CHANGES_REQUESTED** [review]
One small change requested.

File: tests/MongoDB.Driver.Tests/MongoDatabaseImplTests.cs (hunk 1 above)
Comment (BorisDog) [reply tests/MongoDB.Driver.Tests/MongoDatabaseImplTests.cs:1465]: Thanks, done.

**Review by rstam — APPROVED** [review]
LGTM

File: src/MongoDB.Driver/MongoClientBase.cs [hunk 2]
```diff
@@ -1,249 +0,0 @@
...
-using MongoDB.Bson;
-using MongoDB.Driver.Core.Clusters;
-using MongoDB.Driver.Core.Operations;
-
-namespace MongoDB.Driver
-{
-    /// <summary>
-    /// Base class for implementors of <see cref="IMongoClient"/>.
-    /// </summary>
-    public abstract class MongoClientBase : IMongoClient
```
Comment (sanych-sun) [inline src/MongoDB.Driver/MongoClientBase.cs:28]: We should add this to list of breaking changes, together with other deletions and renaming in this PR.

**Review by sanych-sun — APPROVED** [review]
LGTM

**Pull request merged**